Newest versions of vscode-uri (>=1.0.7) break browser frontend

### Description
<!--- 
Replace this comment with a description of the problem.
--> 
Theia browser frontend refuses to start after reinstalling `@theia` dependencies.

### Reproduction Steps
<!-- Describe the issue in as much detail as possible including steps to reproduce. Screenshots and gif screencasts are very helpful. -->
After re-building Docker image with `@theia` packages today (5/29/17), my Theia app would no longer start.

Reverting back from `next` to `latest` versions of `@theia` packages, doesn't solve the problem, suggesting that the issue is with a transitive dependency.

Further, browser console reveals the following and only stacktrace:
```js
index.js:73 Failed to start the frontend application.
(anonymous) @ index.js:73
index.js:75 Error: [UriError]: Scheme is missing: {scheme: "", authority: "", path: "", query: "", fragment: ""}
    at index.js:41
    at t.e (index.js:142)
    at new t (index.js:344)
    at Function.e.parse (index.js:254)
    at new e (uri.ts:29)
    at mini-browser-open-handler.ts:63
    at Object.<anonymous> (mini-browser-open-handler.ts:60)
    at n (bootstrap:19)
    at Object.<anonymous> (mini-browser-frontend-module.ts:30)
    at n (bootstrap:19)
```
where `index.js:41` links to `webpack:///./node_modules/vscode-uri/lib/esm/index.js:41`.

Indeed, it appears that 3 more versions (!) of [microsoft/vscode-uri](https://github.com/microsoft/vscode-uri) were released today - `1.0.7`, `1.0.8`, and `2.0.0`.
 
To confirm this culprit (and provide a workaround), I reverted `vscode-uri` to `1.0.6`, using
```json
  "resolutions": {
    "vscode-uri": "<=1.0.6"
  },
```
in `package.json` (using Yarn). After that, the app starts successfully.

**OS and Theia version:**

All Theia packages are using `next` version (which atm is `0.7.0-next.a963e8b5`, from `yarn.lock`).
The Theia app is built in a Docker container based on `node:10-alpine`.
The host is Mac OS 10.13.6, Chrome 74.0.3729.157.

**Diagnostics:** 
<!-- Provide logs and any other relevant diagnostic information -->

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,12 @@
 # Change Log
 
+## v0.8.0
+
+Breaking changes:
+
+- [core] `scheme` is mandatory for URI
+  - `URI.withoutScheme` is removed, in order to get a path use `URI.path`
+
 ## v0.7.0
 
 - [console] added `Clear Console` command and toolbar item

--- a/configs/warnings.tslint.json
+++ b/configs/warnings.tslint.json
@@ -37,6 +37,7 @@
           "vscode-languageserver-protocol",
           "vscode-languageserver-types",
           "vscode-ws-jsonrpc",
+          "vscode-uri",
           "yargs"
         ]
       ]

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -37,7 +37,7 @@
     "reflect-metadata": "^0.1.10",
     "route-parser": "^0.0.5",
     "vscode-languageserver-types": "^3.10.0",
-    "vscode-uri": "^1.0.1",
+    "vscode-uri": "^1.0.8",
     "vscode-ws-jsonrpc": "^0.0.2-1",
     "ws": "^5.2.2",
     "yargs": "^11.1.0"

--- a/packages/core/src/common/uri.spec.ts
+++ b/packages/core/src/common/uri.spec.ts
@@ -134,8 +134,6 @@ describe('uri', () => {
             const uri = new URI('').withScheme('file').withPath('/foo/bar.txt').withQuery('x=12').withFragment('baz');
             expect(uri.toString(true)).equals('file:///foo/bar.txt?x=12#baz');
 
-            expect(uri.withoutScheme().toString(true)).equals('/foo/bar.txt?x=12#baz');
-
             expect(uri.withScheme('http').toString(true)).equals('http:/foo/bar.txt?x=12#baz');
 
             expect(uri.withoutQuery().toString(true)).equals('file:///foo/bar.txt#baz');

--- a/packages/core/src/common/uri.ts
+++ b/packages/core/src/common/uri.ts
@@ -14,8 +14,12 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
+import { setUriThrowOnMissingScheme } from 'vscode-uri';
 import Uri from 'vscode-uri';
 import { Path } from './path';
+
+// TODO: disable it because of #4487
+setUriThrowOnMissingScheme(false);
 
 export default class URI {
 
@@ -82,13 +86,6 @@ export default class URI {
             scheme
         });
         return new URI(newCodeUri);
-    }
-
-    /**
-     * return this URI without a scheme
-     */
-    withoutScheme(): URI {
-        return this.withScheme('');
     }
 
     /**

--- a/packages/cpp/src/browser/cpp-build-configurations-ui.ts
+++ b/packages/cpp/src/browser/cpp-build-configurations-ui.ts
@@ -78,7 +78,7 @@ export class CppBuildConfigurationChanger implements QuickOpenModel {
         const configurations = this.cppBuildConfigurations.getValidConfigs();
 
         const homeStat = await this.fileSystem.getCurrentUserHome();
-        const home = (homeStat) ? new URI(homeStat.uri).withoutScheme().toString() : undefined;
+        const home = (homeStat) ? new URI(homeStat.uri).path.toString() : undefined;
 
         // Item to create a new build configuration
         items.push(this.createItem);

--- a/packages/debug/package.json
+++ b/packages/debug/package.json
@@ -24,8 +24,7 @@
     "requestretry": "^3.1.0",
     "tar": "^4.0.0",
     "unzip-stream": "^0.3.0",
-    "vscode-debugprotocol": "^1.32.0",
-    "vscode-uri": "^1.0.1"
+    "vscode-debugprotocol": "^1.32.0"
   },
   "publishConfig": {
     "access": "public"

--- a/packages/filesystem/src/node/download/directory-archiver.spec.ts
+++ b/packages/filesystem/src/node/download/directory-archiver.spec.ts
@@ -21,6 +21,7 @@ import { extract } from 'tar-fs';
 import { expect } from 'chai';
 import URI from '@theia/core/lib/common/uri';
 import { MockDirectoryArchiver } from './test/mock-directory-archiver';
+import { FileUri } from '@theia/core/lib/node/file-uri';
 
 // tslint:disable:no-unused-expression
 
@@ -84,8 +85,8 @@ describe('directory-archiver', () => {
         ] as ({ input: string[], expected: Map<string, string[]>, folders?: string[] })[]).forEach(test => {
             const { input, expected, folders } = test;
             it(`should find the common parent URIs among [${input.join(', ')}] => [${Array.from(expected.keys()).join(', ')}]`, async () => {
-                const archiver = new MockDirectoryArchiver(!!folders ? folders.map(u => new URI(u)) : []);
-                const actual = await archiver.findCommonParents(input.map(u => new URI(u)));
+                const archiver = new MockDirectoryArchiver(folders ? folders.map(FileUri.create) : []);
+                const actual = await archiver.findCommonParents(input.map(FileUri.create));
                 expect(asString(actual)).to.be.equal(asString(expected));
             });
         });
@@ -95,7 +96,7 @@ describe('directory-archiver', () => {
             const obj: any = {};
             for (const key of Array.from(map.keys()).sort()) {
                 const values = (map.get(key) || []).sort();
-                obj[key] = `[${values.join(', ')}]`;
+                obj[new URI(key).withScheme('file').toString()] = `[${values.map(v => new URI(v).withScheme('file').toString()).join(', ')}]`;
             }
             return JSON.stringify(obj);
         }

--- a/packages/getting-started/src/browser/getting-started-widget.tsx
+++ b/packages/getting-started/src/browser/getting-started-widget.tsx
@@ -68,7 +68,7 @@ export class GettingStartedWidget extends ReactWidget {
         this.applicationInfo = await this.appServer.getApplicationInfo();
         this.recentWorkspaces = await this.workspaceService.recentWorkspaces();
         this.stat = await this.fileSystem.getCurrentUserHome();
-        this.home = (this.stat) ? new URI(this.stat.uri).withoutScheme().toString() : undefined;
+        this.home = (this.stat) ? new URI(this.stat.uri).path.toString() : undefined;
         this.update();
     }
 

--- a/packages/git/src/browser/blame/blame-decorator.ts
+++ b/packages/git/src/browser/blame/blame-decorator.ts
@@ -36,7 +36,7 @@ export class BlameDecorator implements HoverProvider {
 
     protected registerHoverProvider(uri: string): Disposable {
         if (this.languages.registerHoverProvider) {
-            return this.languages.registerHoverProvider([{ pattern: new URI(uri).withoutScheme().toString() }], this);
+            return this.languages.registerHoverProvider([{ pattern: new URI(uri).path.toString() }], this);
         }
         return Disposable.NULL;
     }

--- a/packages/git/src/browser/dirty-diff/dirty-diff-manager.ts
+++ b/packages/git/src/browser/dirty-diff/dirty-diff-manager.ts
@@ -237,8 +237,8 @@ export class DirtyDiffModel implements Disposable {
     }
 
     protected async isInGitRepository(repository: Repository): Promise<boolean> {
-        const modelUri = this.editor.uri.withoutScheme().toString();
-        const repoUri = new URI(repository.localUri).withoutScheme().toString();
+        const modelUri = this.editor.uri.withScheme('file').toString();
+        const repoUri = new URI(repository.localUri).withScheme('file').toString();
         return modelUri.startsWith(repoUri) && this.previousRevision.isVersionControlled();
     }
 

--- a/packages/git/src/browser/git-resource.ts
+++ b/packages/git/src/browser/git-resource.ts
@@ -58,7 +58,7 @@ export class GitResourceResolver implements ResourceResolver {
     }
 
     async getRepository(uri: URI): Promise<Repository | undefined> {
-        const uriWithoutScheme = uri.withoutScheme();
+        const fileUri = uri.withScheme('file');
         const repositories = this.repositoryProvider.allRepositories;
         // The layout restorer might ask for the known repositories this point.
         if (repositories.length === 0) {
@@ -72,9 +72,9 @@ export class GitResourceResolver implements ResourceResolver {
         const sortedRepositories = repositories.sort((a, b) => b.localUri.length - a.localUri.length);
         for (const repository of sortedRepositories) {
             const localUri = new URI(repository.localUri);
-            // make sure that localUri of repository has no scheme.
-            const localUriStr = localUri.withoutScheme().toString();
-            if (uriWithoutScheme.toString().startsWith(localUriStr)) {
+            // make sure that localUri of repository has file scheme.
+            const localUriStr = localUri.withScheme('file').toString();
+            if (fileUri.toString().startsWith(localUriStr)) {
                 return { localUri: localUriStr };
             }
         }

--- a/packages/mini-browser/src/browser/location-mapper-service.ts
+++ b/packages/mini-browser/src/browser/location-mapper-service.ts
@@ -136,7 +136,7 @@ export class FileLocationMapper implements LocationMapper {
         if (uri.scheme !== 'file') {
             throw new Error(`Only URIs with 'file' scheme can be mapped to an URL. URI was: ${uri}.`);
         }
-        let rawLocation = uri.withoutScheme().toString();
+        let rawLocation = uri.path.toString();
         if (rawLocation.charAt(0) === '/') {
             rawLocation = rawLocation.substr(1);
         }

--- a/packages/monaco/src/browser/monaco-workspace.ts
+++ b/packages/monaco/src/browser/monaco-workspace.ts
@@ -16,6 +16,7 @@
 
 // tslint:disable:no-null-keyword
 
+import Uri from 'vscode-uri';
 import { injectable, inject, postConstruct } from 'inversify';
 import { ProtocolToMonacoConverter, MonacoToProtocolConverter, testGlob } from 'monaco-languageclient';
 import URI from '@theia/core/lib/common/uri';
@@ -220,11 +221,11 @@ export class MonacoWorkspace implements lang.Workspace {
 
     createFileSystemWatcher(globPattern: string, ignoreCreateEvents?: boolean, ignoreChangeEvents?: boolean, ignoreDeleteEvents?: boolean): lang.FileSystemWatcher {
         const disposables = new DisposableCollection();
-        const onDidCreateEmitter = new lang.Emitter<monaco.Uri>();
+        const onDidCreateEmitter = new lang.Emitter<Uri>();
         disposables.push(onDidCreateEmitter);
-        const onDidChangeEmitter = new lang.Emitter<monaco.Uri>();
+        const onDidChangeEmitter = new lang.Emitter<Uri>();
         disposables.push(onDidChangeEmitter);
-        const onDidDeleteEmitter = new lang.Emitter<monaco.Uri>();
+        const onDidDeleteEmitter = new lang.Emitter<Uri>();
         disposables.push(onDidDeleteEmitter);
         disposables.push(this.fileSystemWatcher.onFilesChanged(changes => {
             for (const change of changes) {
@@ -239,8 +240,7 @@ export class MonacoWorkspace implements lang.Workspace {
                     continue;
                 }
                 const uri = change.uri.toString();
-                // tslint:disable-next-line:no-any
-                const { codeUri } = (change.uri as any);
+                const codeUri = change.uri['codeUri'];
                 if (testGlob(globPattern, uri)) {
                     if (fileChangeType === FileChangeType.ADDED) {
                         onDidCreateEmitter.fire(codeUri);

--- a/packages/plugin-ext-vscode/package.json
+++ b/packages/plugin-ext-vscode/package.json
@@ -5,8 +5,7 @@
   "dependencies": {
     "@theia/core": "^0.7.0",
     "@theia/plugin": "^0.7.0",
-    "@theia/plugin-ext": "^0.7.0",
-    "vscode-uri": "^1.0.1"
+    "@theia/plugin-ext": "^0.7.0"
   },
   "publishConfig": {
     "access": "public"

--- a/packages/plugin-ext/package.json
+++ b/packages/plugin-ext/package.json
@@ -26,8 +26,7 @@
     "lodash.clonedeep": "^4.5.0",
     "ps-tree": "1.1.0",
     "uuid": "^3.2.1",
-    "vscode-debugprotocol": "^1.32.0",
-    "vscode-uri": "^1.0.1"
+    "vscode-debugprotocol": "^1.32.0"
   },
   "publishConfig": {
     "access": "public"

--- a/packages/plugin-ext/src/plugin/preferences/configuration.spec.ts
+++ b/packages/plugin-ext/src/plugin/preferences/configuration.spec.ts
@@ -201,7 +201,7 @@ describe('Configuration:', () => {
             configuration = new Configuration(
                 defaultConfiguration, userConfiguration, workspaceConfiguration, folderConfigurations
             );
-            const resource = URI.revive({ path: project });
+            const resource = URI.revive({ path: project, scheme: '', authority: '', query: '', fragment: '' });
             inspect = configuration.inspect(propertyName, workspace, resource);
         });
 

--- a/packages/plugin-ext/src/plugin/workspace.ts
+++ b/packages/plugin-ext/src/plugin/workspace.ts
@@ -161,7 +161,7 @@ export class WorkspaceExtImpl implements WorkspaceExt {
         }
 
         return this.proxy.$startFileSearch(includePattern, includeFolderUri, excludePatternOrDisregardExcludes, maxResults, token)
-            .then(data => Array.isArray(data) ? data.map(URI.revive) : []);
+            .then(data => Array.isArray(data) ? data.map(uri => URI.revive(uri)) : []);
     }
 
     createFileSystemWatcher(globPattern: theia.GlobPattern, ignoreCreateEvents?: boolean, ignoreChangeEvents?: boolean, ignoreDeleteEvents?: boolean): theia.FileSystemWatcher {

--- a/packages/preferences/src/browser/preferences-tree-widget.ts
+++ b/packages/preferences/src/browser/preferences-tree-widget.ts
@@ -428,10 +428,10 @@ export class PreferencesEditorsContainer extends DockPanel {
 
         let uri = preferenceUri;
         if (preferenceUri.scheme === UserStorageUri.SCHEME && homeUri) {
-            uri = homeUri.resolve(THEIA_USER_STORAGE_FOLDER).resolve(preferenceUri.withoutScheme().path);
+            uri = homeUri.resolve(THEIA_USER_STORAGE_FOLDER).resolve(preferenceUri.path);
         }
         return homeUri
-            ? FileSystemUtils.tildifyPath(uri.path.toString(), homeUri.withoutScheme().toString())
+            ? FileSystemUtils.tildifyPath(uri.path.toString(), homeUri.path.toString())
             : uri.path.toString();
     }
 }

--- a/packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx
+++ b/packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx
@@ -353,12 +353,12 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
 
     protected getFileNodesByUri(uri: URI): SearchInWorkspaceFileNode[] {
         const nodes: SearchInWorkspaceFileNode[] = [];
-        const path = uri.withoutScheme().toString();
+        const fileUri = uri.withScheme('file').toString();
         for (const rootFolderNode of this.resultTree.values()) {
             const rootUri = new URI(rootFolderNode.path).withScheme('file');
             if (rootUri.isEqualOrParent(uri)) {
                 for (const fileNode of rootFolderNode.children) {
-                    if (fileNode.fileUri === path) {
+                    if (fileNode.fileUri === fileUri) {
                         nodes.push(fileNode);
                     }
                 }
@@ -650,7 +650,6 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
 
     protected async createReplacePreview(node: SearchInWorkspaceFileNode): Promise<URI> {
         const fileUri = new URI(node.fileUri).withScheme('file');
-        const uri = fileUri.withoutScheme().toString();
         const resource = await this.fileResourceResolver.resolve(fileUri);
         const content = await resource.readContents();
 
@@ -663,7 +662,7 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
             lines[l.line - 1] = start + this._replaceTerm + end;
         });
 
-        return new URI(uri).withScheme(MEMORY_TEXT).withQuery(lines.join('\n'));
+        return fileUri.withScheme(MEMORY_TEXT).withQuery(lines.join('\n'));
     }
 
     protected decorateEditor(node: SearchInWorkspaceFileNode | undefined, editorWidget: EditorWidget) {

--- a/packages/task/src/node/task-server.slow-spec.ts
+++ b/packages/task/src/node/task-server.slow-spec.ts
@@ -130,7 +130,7 @@ describe('Task server / back-end', function () {
 
     it('task is executed successfully with cwd as a file URI', async function () {
         const command = isWindows ? commandShortrunningindows : commandShortRunning;
-        const config = createProcessTaskConfig('shell', FileUri.fsPath(command), [], FileUri.create(wsRoot).toString());
+        const config = createProcessTaskConfig('shell', command, [], FileUri.create(wsRoot).toString());
         const taskInfo: TaskInfo = await taskServer.run(config, wsRoot);
 
         const p = checkSuccessfullProcessExit(taskInfo, taskWatcher);
@@ -140,7 +140,7 @@ describe('Task server / back-end', function () {
 
     it('task is executed successfully using terminal process', async function () {
         const command = isWindows ? commandShortrunningindows : commandShortRunning;
-        const taskInfo: TaskInfo = await taskServer.run(createProcessTaskConfig('shell', FileUri.fsPath(command), []), wsRoot);
+        const taskInfo: TaskInfo = await taskServer.run(createProcessTaskConfig('shell', command, []), wsRoot);
 
         const p = checkSuccessfullProcessExit(taskInfo, taskWatcher);
 

--- a/packages/terminal/src/browser/terminal-frontend-contribution.ts
+++ b/packages/terminal/src/browser/terminal-frontend-contribution.ts
@@ -349,7 +349,7 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
     protected async selectTerminalCwd(): Promise<string | undefined> {
         const roots = this.workspaceService.tryGetRoots();
         return this.quickPick.show(roots.map(
-            ({ uri }) => ({ label: this.labelProvider.getLongName(new URI(uri).withoutScheme()), value: uri })
+            ({ uri }) => ({ label: this.labelProvider.getLongName(new URI(uri)), value: uri })
         ), { placeholder: 'Select current working directory for new terminal' });
     }
 

--- a/packages/workspace/src/browser/quick-open-workspace.ts
+++ b/packages/workspace/src/browser/quick-open-workspace.ts
@@ -38,7 +38,7 @@ export class QuickOpenWorkspace implements QuickOpenModel {
     async open(workspaces: string[]): Promise<void> {
         this.items = [];
         const homeStat = await this.fileSystem.getCurrentUserHome();
-        const home = (homeStat) ? new URI(homeStat.uri).withoutScheme().toString() : undefined;
+        const home = (homeStat) ? new URI(homeStat.uri).path.toString() : undefined;
         let tempWorkspaceFile: URI | undefined;
         if (home) {
             tempWorkspaceFile = getTemporaryWorkspaceFileUri(new URI(home));

